### DISCUSSION TITLE
LESSON 3 fundMe getConversionRate  videoTime 4:17:24   DeclarationError: Undeclared Identifier

### DISCUSSION THREAD

**Discussion by franclino** [smartcontractkit/full-blockchain-solidity-course-py#1813] [Q&A] [answered]
Hi guys. I´m getting this error 
<img width="275" alt="Screenshot 2022-12-21 at 16 24 14" src="https://user-images.githubusercontent.com/71766287/208940837-69e78bda-1f82-45a3-90df-068760b20667.png">

my code is this basically copying everything Patrick did:

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.8;

import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

contract FundMe {

    uint256 public minimumUsd = 50;

    function fund() public payable{
        //want to be able to set a minimum fund amount in USD
        //1.How do we send ETH to this contract?

        require(msg.value >= minimumUsd, "Didnt send enough!");
    }

    function getPrice() public view returns (uint256) {
        // ABI
        // Address 0xD4a33860578De61DBAbDc8BFdb98FD742fA7028e
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0xD4a33860578De61DBAbDc8BFdb98FD742fA7028e);
        (,int256 price,,,) = priceFeed.latestRoundData();
        //ETH in terms of USD
        //3000.00000000
        return uint256(price * 1e10); //1**10 00000000
    }

    function getVersion() public view returns (uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0xD4a33860578De61DBAbDc8BFdb98FD742fA7028e);
        return priceFeed.version();
    }

    function getConversionRate(uint256 ethAmount) public view returns (uint256) {
        uint256 ethPrice = getPrice();
        uint256 ethAmount = (ethPrice * ethAmount) / 1e18; 
    return ethAmountInUsd;
    }

    //function withdraw(){}
}

i cannot find my mistake could you help? 

Thank you,

franclino

**Comment by Niferu** [comment, marked as answer]
Hello @franclino 

You have used incorrect variable in `getConversionRate()` function, you have just used `ethAmount` in wrong place, function should look like below:

```solidity
    function getConversionRate(uint256 ethAmount) public view returns (uint256) {
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1e18; 
        return ethAmountInUsd;
    }
```

**Reply by cromewar** [reply]
This must be the solution, amazing @Neftyr
Does this make sense to you @franclino?

**Reply by franclino** [reply]
Hi! yes i see it thank you so much!

**Reply by cromewar** [reply]
setting this up as answer then!
